woocommerce/woocommerce
# [Grouped products] Product list

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #35146



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a `Grouped` product from the classic editor
2. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
3. Make sure feature `product-grouped` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
4. If you try to edit the product created in point 1 you should be redirected to the new product editor.
5. From there under the `General` tab, a new `Products in this group` section should be shown 
<img width="707" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/4a02e90b-1e63-48bc-889b-5057fe1f16de">

6. The Pricing and Shipping tabs should be hidden.
7. When empty, the grouped products card has an empty state with an illustration 
<img width="718" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/0527cfe7-7cfa-404b-9ea6-f977557d5a81">

8. When the user clicks `Add products`, we show a modal. 
<img width="787" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/38c11729-4f50-4bb6-976d-7942461bd24a">

9. When the user clicks the search field, we immediately display the list of all existing products. It is sorted alphabetically. Clicking an item closes the search and adds the product to the list. 
<img width="787" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/ff8ac985-88d5-4556-ab22-605056ef00dd">

10. The user can search the list by product name. 
<img width="696" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/3080394f-e3b7-469c-98ba-a7f1b19937c0">

11. The user can add multiple products and delete products directly in the modal. 
<img width="677" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/ee2715b5-b5fd-41ff-8323-a74897ba6fed">

12. When the user clicks `Add`, we close the modal and display the list of added products in the product form.
13. The list works the same as the one in the modal, but it has more details, like the price and current stock.
14. Clicking the product name will open it in the editing view in a new tab.
15. If the name or SKU extends beyond this width, we truncate the text.
16. Clicking the arrow button will open the product's page in a new tab.
17. Clicking the X button will remove the product from the group.
18. Clicking Add products will re-open the modal (it'll be empty).
19. When the user removes all products, we show the empty state again.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
